src/core/meshcop/leader.hpp

```diff
@@ -26,11 +26,131 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if OPENTHREAD_MTD
-#include "leader_mtd.hpp"
-#elif OPENTHREAD_FTD
-#include "leader_ftd.hpp"
-#else
-#error "Please define OPENTHREAD_MTD=1 or OPENTHREAD_FTD=1"
-#endif
+/**
+ * @file
+ *   This file includes definitions for a MeshCoP Leader.
+ */
+
+#ifndef MESHCOP_LEADER_HPP_
+#define MESHCOP_LEADER_HPP_
+
+#include <coap/coap_client.hpp>
+#include <coap/coap_server.hpp>
+#include <common/timer.hpp>
+#include <meshcop/tlvs.hpp>
+#include <net/udp6.hpp>
+#include <thread/mle.hpp>
+
+namespace Thread {
+namespace MeshCoP {
+
+OT_TOOL_PACKED_BEGIN
+class CommissioningData
+{
+public:
+    uint8_t GetLength(void) const {
```

Comment:
The change in #1550 reduced the MTD build size by about 2.2KB (on an internal hw platform). 

That said, I fully agree with @DuaneEllis-TI. We should consider inlining only simple methods. #1550 inlined the simple getters/setters which were basically doing member variable access.